.github/workflows/publish-docker-snapshot.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Publish Snapshot Docker Image
+on: create
```

Comment:
I gather GitHub still haven't added a `tag` create filter to this like they have for `release`?

Reply:
Yeah there isn't a specific event for tag creation. `create` is fired when a new ref is created. A ref can either be a tag or a branch, so the way to run an action on tag create is

```yaml
on: create
jobs:
  build:
    if: github.event.ref_type == 'tag' 
```